[WIP] Alien Skirmisher

ephemeralstoryteller opened the pull request:
#### Summary

SUMMARY: Adds new unit for the light alien patrols, the alien skirmisher


#### Purpose of change

Adds a standard skirmishing and recon unit to the alien roster, one of the first cyborgs a player will see.

#### Describe alternatives you've considered

_Considerations for discussion:_

**The Skirmisher has some personality**. It has trophies on its gear. I wanted to cement it as a memorable enemy that might truly view the player as "prey", and to solidify the general alien view of humanity. Should this be kept, or should the aliens be a largely stoic, purely functionalist force?

**Should the Skirmisher have a special attack?**. I want to give it the "impale" special attack, or at least something very similar, to make it a credible threat that shouldn't be engaged in its preferable range, ie melee. Would that be too dangerous? Alternatively, what might be more thematic? A thermal lance attack?

**Are its stats too low?**. I just need to test this at some point, but I wanted feedback now.

Low priority consideration, may be better for another PR: If the Skirmisher should receive a ranged weapon when monsters are less likely to shoot one another, or should that be a separate unit?

#### Testing

None yet

#### Additional context

Requires #48150 from master due to calling that harvest entry.

ephemeralstoryteller committed:
Update alien_cyborgs.json

ephemeralstoryteller committed:
lint

mlangsdorf commented on data/mods/Dark-Skies-Above/monsters/alien_cyborgs.json:
```diff
@@ -286,5 +286,34 @@
+    "morale": 100,
+    "color": "light_gray",
+    "dodge": 1,
+    "armor_bash": 15,
```
```suggestion
    "armor_bash": 25,
```
I'd like to make it hard to engage with crushing weapons, because the current weapons meta favors crushing or piercing weapons too much.

mlangsdorf commented on data/mods/Dark-Skies-Above/monsters/alien_cyborgs.json:
```diff
@@ -286,5 +286,34 @@
+    "color": "light_gray",
+    "dodge": 1,
+    "armor_bash": 15,
+    "armor_cut": 6,
```
We'll need to specify its pierce resistance specifically, because the default 80% of cut is way too low.

`+    "armor_stab": 25,`

should do it.

Also,
`+    "armor_fire": 15,`

so people can't just kill them with molotovs.

mlangsdorf commented on data/mods/Dark-Skies-Above/monsters/alien_cyborgs.json:
```diff
@@ -286,5 +286,34 @@
+    "id": "mon_dks_skirmsoldier",
+    "type": "MONSTER",
+    "name": { "str": "alien skirmisher" },
+    "description": "A slouched figure, slightly smaller than an average adult human.  Beneath layers of full-body combat armor and a full suite of cybernetic modifications are glimpses of rubbery, abscessed flesh.  It is capable of short, boosted jumps using a device on its back, closing gaps to cause havoc with its wicked integrated melee weapons.  Severed organic parts dangle from its equipment ...
```
```suggestion
    "description": "A slouched bipedal figure, slightly smaller than an average adult human.  Beneath layers of full-body combat armor and a full suite of cybernetic modifications are glimpses of rubbery, abscessed flesh.  It is capable of short, boosted jumps using a device on its back, closing gaps to cause havoc with its wicked integrated melee weapons.  Severed organic parts dangle from its equipment, morbid trophies from past battles.",
```

I had been envisioning these as manhacks, but creepy little cyborgs with battle trophies are great.  I just think the fact that they're bipedal needs to be made clear.

ephemeralstoryteller replied:
Bipedal is a much better word than "humanoid", which was in my original draft and removed because I felt like it was a bit too explicit. I like the players using their imagination a little as to the exacts of its form and how it attacks.

mlangsdorf reviewed (commented):
I generally like it.  I think instead of a thermal lance, they should use some kind of electric jolt weapon.  Either a taser or a short range lightning bolt kind of thing like the shocker zombies use.

ephemeralstoryteller committed:
Update data/mods/Dark-Skies-Above/monsters/alien_cyborgs.json

Co-authored-by: Mark Langsdorf <mark.langsdorf@gmail.com>

mlangsdorf commented:
Doh, forgot to answer your questions:

The numbers look good - I think the HP and speed are about perfect.

The immediate response skirmishers should be melee troops, not ranged.  They have can have an upgrade version that has a ranged attack and is probably the thin g that gets summoned when they die.

I'll try to work on a death function that summons more monsters this week, but I don't know if I'll get that done.  Still, we can update these later.

ephemeralstoryteller committed:
bash armor up

Co-authored-by: Mark Langsdorf <mark.langsdorf@gmail.com>

ephemeralstoryteller commented:
> I generally like it. I think instead of a thermal lance, they should use some kind of electric jolt weapon. Either a taser or a short range lightning bolt kind of thing like the shocker zombies use.

Electricity sounds good. Should they then be immune to electricity themselves? 

> The immediate response skirmishers should be melee troops, not ranged. They have can have an upgrade version that has a ranged attack and is probably the thin g that gets summoned when they die.

:+1:  I have a few ideas for the upgrade, but we'll get there when we get there.


re: in general making the aliens a bit more creepy and with implied personality

Ultimately I think this might be the move. The soulessness of what's happening can be more behind-the-curtain in the form of the player one day realizing this was all done at the behest of a min-maxing AI. Enemies that are practical and faceless in every way are ultimately a bit bland, hence why XCOM surely chose the host of different designs it did.

ephemeralstoryteller committed:
stab and fire armor upped

also specifies that the combat armor is woven, to hint towards being weaker to cut damage

ephemeralstoryteller committed:
Merge branch 'dsa-patrol_drones' of https://github.com/ephemeralstoryteller/Cataclysm-DDA into dsa-patrol_drones

ephemeralstoryteller committed:
adds attacks to skirmisher

gives it a melee shock and an overall decent melee ability

mlangsdorf commented:
I don't think the low-level aliens should be resistant to electricity.  Emissaries and AHMRs should be more resistant to it, but CDDA only has so many damage types and the aliens should be somewhat vulnerable to cracked/homebrew weapons based on their own weapons.

ephemeralstoryteller committed:
makes skirm's shock more ambiguous

might be a physical attack, might be a short-range burst. intended to let the player fill in the gaps with their imagination more in order to preserve the "alien" feel

ephemeralstoryteller committed:
Update alien_cyborgs.json

raises melee skill by 1 to indicate a relatively skilled opponent.
adds "swarms", as they work better in groups.

ephemeralstoryteller committed:
skirm changes

-makes their leap slightly less quick
-changes naming scheme to be congruent with other monsters

ephemeralstoryteller committed:
skirm summoning

-makes a spell for summoning skirmishers
